src/cargo/util/auth/mod.rs

```
@@ -454,7 +453,10 @@ fn credential_action(
         tracing::debug!("attempting credential provider: {args:?}");
         let provider: Box<dyn Credential> = match process {
             "cargo:token" => Box::new(TokenCredential::new(config)),
-            "cargo:paseto" => Box::new(PasetoCredential::new(config)),
+            "cargo:paseto" if config.cli_unstable().asymmetric_token => {
+                Box::new(PasetoCredential::new(config))
+            }
+            "cargo:paseto" => bail!("cargo:paseto requires -Zasymmetric-token"),
```

We could also decide to silently skip it rather than error here.